pmd-java/src/main/resources/rulesets/java/android.xml

```diff
@@ -9,116 +9,9 @@ These rules deal with the Android SDK, mostly related to best practices.
-[ancestor::ClassOrInterfaceDeclaration[ExtendsList/ClassOrInterfaceType[
-  typeof(@Image, 'android.app.Activity', 'Activity') or
-  typeof(@Image, 'android.app.Application', 'Application') or
-  typeof(@Image, 'android.app.Service', 'Service')
-]]]
-]]>
-        </value>
-      </property>
-    </properties>
-    <example><![CDATA[
-public class DummyActivity extends Activity {
-    public void onCreate(Bundle bundle) {
-        // missing call to super.onCreate(bundle)
-        foo();
-    }
-}
-]]>
-    </example>
-  </rule>
-
-  <rule name="CallSuperLast"
-      since="4.2.5"
-      language="java"
-      message="super should be called at the end of the method"
-      class="net.sourceforge.pmd.lang.rule.XPathRule"
-      externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_android.html#callsuperlast">
-    <description>Super should be called at the end of the method</description>
-    <priority>3</priority>
-    <properties>
-      <property name="xpath">
-        <value>
-          <![CDATA[
-//MethodDeclaration[MethodDeclarator[
-  @Image='finish' or
-  @Image='onDestroy' or
-  @Image='onPause' or
-  @Image='onSaveInstanceState' or
-  @Image='onStop' or
-  @Image='onTerminate'
-  ]]
-   /Block/BlockStatement[last()]
-    [not(Statement/StatementExpression/PrimaryExpression[./PrimaryPrefix[@SuperModifier='true']]/PrimarySuffix[@Image= ancestor::MethodDeclaration/MethodDeclarator/@Image])]
-[ancestor::ClassOrInterfaceDeclaration[ExtendsList/ClassOrInterfaceType[
-  typeof(@Image, 'android.app.Activity', 'Activity') or
-  typeof(@Image, 'android.app.Application', 'Application') or
-  typeof(@Image, 'android.app.Service', 'Service')
-]]]
-]]>
-        </value>
-      </property>
-    </properties>
-    <example>
-      <![CDATA[
-public class DummyActivity extends Activity {
-    public void onPause() {
-        foo();
-        // missing call to super.onPause()
-    }
-}
-]]>
-    </example>
-  </rule>
-
-  <rule name="DoNotHardCodeSDCard"
-      since="4.2.6"
-      language="java"
-      message="Do not hardcode /sdcard."
-      class="net.sourceforge.pmd.lang.rule.XPathRule"
-      externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_android.html#donothardcodesdcard">
-    <description>Use Environment.getExternalStorageDirectory() instead of "/sdcard"</description>
-    <priority>3</priority>
-    <properties>
-      <property name="xpath">
-        <value>//Literal[starts-with(@Image,'"/sdcard')]</value>
-      </property>
-    </properties>
-    <example>
-      <![CDATA[
-public class MyActivity extends Activity {
-    protected void foo() {
-        String storageLocation = "/sdcard/mypackage";   // hard-coded, poor approach
-
-       storageLocation = Environment.getExternalStorageDirectory() + "/mypackage"; // preferred approach
-    }
-}
-]]>
-    </example>
-  </rule>
+    <!-- Rules, that have been moved into a category -->
+    <rule ref="java-errorprone/CallSuperFirst" deprecated="true" />
```

Comment:
It will just log, that the rule has been moved - the deprecated flag is on the rule-ref and not on the rule itself.
However, right now, we wouldn't generate documentation from the old rulesets, only from the new categories... so the warning is visible only when using the android ruleset (hopefully...).